src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,186 @@
+package duke.task;
+
+import duke.Parser;
+import duke.Ui;
+
+import java.time.LocalDateTime;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Encapsulates an arraylist of Task with the various methods
+ */
+public class TaskList {
+    private ArrayList<Task> list;
+
+    public TaskList(ArrayList<Task> list) {
+        this.list = list;
+    }
+
+    public ArrayList<Task> getList() {
+        return this.list;
+    }
+
+    /**
+     * Nicely prints out all the task
+     */
+    public String list() {
```

Comment:
I think another name might be more appropriate here, since list is not very intuitive. 
```suggestion
    public String listAllTasks() {
```